dataline-scheduler/src/main/java/io/dataline/scheduler/WorkerRun.java

```diff
@@ -48,101 +43,27 @@
  * clarify it later. the main benefit is of this class is that it gives us some type safety when
  * working with workers. you can probably make an argument that this class should not have access to
  * the db.
- *
- * @param <InputType> - the type that the worker consumes.
- * @param <OutputType> - the type that the worker outputs.
  */
-public class WorkerRun<InputType, OutputType> implements Runnable {
+public class WorkerRun implements Callable<OutputAndStatus<?>> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(WorkerRun.class);
 
-  private final long jobId;
   private final Path jobRoot;
-  private final InputType input;
-  private final Worker<InputType, OutputType> worker;
-  private final BasicDataSource connectionPool;
+  private final CheckedSupplier<OutputAndStatus<?>, Exception> outputSupplier;
 
-  public WorkerRun(long jobId,
-                   Path jobRoot,
-                   InputType input,
-                   Worker<InputType, OutputType> worker,
-                   BasicDataSource connectionPool) {
-    this.jobId = jobId;
+  public <InputType> WorkerRun(final Path jobRoot,
+                               final InputType input,
+                               final Worker<InputType, ?> worker) {
     this.jobRoot = jobRoot;
-    this.input = input;
-    this.worker = worker;
-    this.connectionPool = connectionPool;
+    this.outputSupplier = () -> worker.run(input, jobRoot);
```

Comment:
why do you prefer this as opposed to making `input` and `jobRoot` fields of the class and calling `worker.run(...)` in the call method?

Reply:
if you do want to keep this structure can we call this field something different than `outputSupplier`? maybe `runner` or something. `outputSupplier` seems slightly misleading. doesn't really convey to me that a whole worker is being run here.

Reply:
I removed the unnecessary generic of `WorkerRun` while still keeping the input type safety. Creating this supplier was the only way to keep the input type. I could use a reference otherwise instead of using the supplier.